pkg/validators/am0010_k8s_resource_names.go

```diff
@@ -20,215 +18,152 @@ func init() {
 	}
 
 	annotations := *mb.AddonMeta.CommonAnnotations
 
-	result := make([]failureMsg, 0, len(annotations))
+	result := make([]string, 0, len(annotations))
 
-	if msgs := areValidAnnotationNames(keys(annotations)...); len(msgs) > 0 {
-		result = append(result, prefixedFailureMsg(
-			"commonAnnotations", joinFailureMsgs(msgs...),
-		))
+	if msgs := utils.AreValidk8sAnnotationNames(keys(annotations)...); len(msgs) > 0 {
+		for _, msg := range msgs {
+			result = append(result, prefixedFailureMsg(
+				"commonAnnotations", msg,
+			))
+		}
 	}
 
 	return result
 }
 
-func validateCommonLabels(mb types.MetaBundle) []failureMsg {
+func validateCommonLabels(mb types.MetaBundle) []string {
 	if mb.AddonMeta.CommonLabels == nil {
-		return []failureMsg{}
+		return []string{}
 	}
 
 	labels := *mb.AddonMeta.CommonLabels
 
-	result := make([]failureMsg, 0, len(labels))
+	result := make([]string, 0, len(labels))
 
-	if msgs := areValidLabelNames(keys(labels)...); len(msgs) > 0 {
-		result = append(result, prefixedFailureMsg(
-			"commonLabels", joinFailureMsgs(msgs...),
-		))
+	if msgs := utils.AreValidk8sLabelNames(keys(labels)...); len(msgs) > 0 {
+		for _, msg := range msgs {
+			result = append(result, prefixedFailureMsg(
+				"commonLabels", msg,
+			))
+		}
 	}
 
 	return result
 }
 
-type k8sName string
-
-const (
-	k8sNameAnnotation = k8sName("annotation")
-	k8sNameLabel      = k8sName("label")
-	k8sNameNamespace  = k8sName("namespace")
-)
-
-func areValidAnnotationNames(names ...string) []failureMsg {
-	return areValidNames(k8sNameAnnotation, names...)
-}
-
-func areValidLabelNames(names ...string) []failureMsg {
-	return areValidNames(k8sNameLabel, names...)
-}
-
-func areValidNamespaceNames(names ...string) []failureMsg {
-	return areValidNames(k8sNameNamespace, names...)
-}
+func validatePagerDutyFields(mb types.MetaBundle) []string {
+	pd := mb.AddonMeta.PagerDuty
+	if pd == nil {
+		return []string{}
+	}
 
-func areValidNames(nameType k8sName, names ...string) []failureMsg {
-	validationFunc := k8sNameToValidationFunc(nameType)
+	var msgs []string
 
-	msgs := make([]failureMsg, 0, len(names))
+	if msg := utils.IsValidk8sNamespaceName(pd.SecretNamespace); msg != "" {
+		msgs = append(msgs, msg)
+	}
 
-	for _, name := range names {
-		if msg := validationFunc(name); !msg.IsEmpty() {
-			msgs = append(msgs, msg)
-		}
+	if msg := utils.IsValidk8sSecretName(pd.SecretName); msg != "" {
+		msgs = append(msgs, msg)
 	}
 
 	return msgs
 }
 
-func k8sNameToValidationFunc(nameType k8sName) func(string) failureMsg {
-	switch nameType {
-	case k8sNameAnnotation:
-		return isValidAnnotationName
-	case k8sNameLabel:
-		return isValidLabelName
-	case k8sNameNamespace:
-		return isValidNamespaceName
-	default:
-		// panic is preffered here since this indicates a programming error
-		panic(fmt.Sprintf("no validation function defined for '%v'", nameType))
+func validateBundleFields(mb types.MetaBundle) []string {
+	bp := mb.AddonMeta.BundleParameters
```

Comment:
The `BundleParameters` is only used for building bundles from within `managed-tenants`, and will be deprecated as soon as we migrate all bundles to `managed-tenants-bundles`.

I would discourage investing effort in validating them. The migration is ongoing https://issues.redhat.com/browse/MTSRE-162. I suppose we should add a deprecation comment to `pkg/mtsre/v1/addons.go` and `api/v1alpha1/addonmetadata_types.go`

Reply:
I can put in the deprecation notices.